lib/vcloud/core/vm.rb

```diff
@@ -200,6 +223,26 @@ def virtual_hardware_section
         vcloud_attributes[:'ovf:VirtualHardwareSection'][:'ovf:Item']
       end
 
+      def get_media_id_by_name(media_name)
+        media_id = get_media_href_by_name(media_name)
+        media_id.split("/").last
+      end
+
+      def get_media_href_by_name(media_name)
+        q = Vcloud::Core::QueryRunner.new 
+        vdc_results = q.run('vApp', :filter => "name==#{vapp_name}")
```

Comment:
I'm not sure this query will work because `vapp_name` is undefined in this method.